### PR TITLE
Cache reserved keyword list

### PR DESCRIPTION
**Pull request by Slamdunk**

### CODE REVIEW

File: src/Fixer/FunctionNotation/PhpdocToReturnTypeFixer.php [hunk 1]
```diff
@@ -343,4 +346,27 @@ private function findReturnAnnotations(Tokens $tokens, $index)
...
     }
+
+    /**
+     * Test if a given return type is a reserved PHP keyword.
+     *
+     * @param string $returnType
+     *
+     * @return bool
+     */
+    private function isReservedKeyword($returnType)
```
Comment (kubawerlos) [inline src/Fixer/FunctionNotation/PhpdocToReturnTypeFixer.php:357]: I wonder if we can move this to some class in `PhpCsFixer\Tokenizer\Analyzer` (even new one) as for `PhpdocToParamTypeFixer` we would have to have the same check, WDYT?

File: src/Fixer/FunctionNotation/PhpdocToReturnTypeFixer.php [hunk 2]
```diff
@@ -251,9 +256,7 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
                 continue;
             }
 
-            try {
-                Tokens::fromCode(sprintf('<?php function f():%s {}', $returnType));
-            } catch (\ParseError $e) {
+            if ($this->isReservedKeyword($returnType)) {
```
[the remark this answers is not in the document]
Comment (Slamdunk) [reply src/Fixer/FunctionNotation/PhpdocToReturnTypeFixer.php:259]: Seems legit

File: src/Fixer/FunctionNotation/PhpdocToReturnTypeFixer.php [hunk 3]
```diff
@@ -343,4 +346,27 @@ private function findReturnAnnotations(Tokens $tokens, $index)
...
+    /**
+     * Test if a given return type is a reserved PHP keyword.
+     *
+     * @param string $returnType
+     *
+     * @return bool
+     */
+    private function isReservedKeyword($returnType)
+    {
+        if (! array_key_exists($returnType, $this->reservedKeywordCache)) {
```
[the remark this answers is not in the document]
Comment (Slamdunk) [reply src/Fixer/FunctionNotation/PhpdocToReturnTypeFixer.php:359]: Sorry for the superficial code, I wrote the code in the browser, I was in a hurry, so no

File: src/Fixer/FunctionNotation/PhpdocToReturnTypeFixer.php (hunk 1 above)
Comment (Slamdunk) [reply src/Fixer/FunctionNotation/PhpdocToReturnTypeFixer.php:357]: Good idea

File: src/Fixer/FunctionNotation/PhpdocToReturnTypeFixer.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (kubawerlos) [reply src/Fixer/FunctionNotation/PhpdocToReturnTypeFixer.php:259]: This is done.

File: src/Fixer/FunctionNotation/PhpdocToReturnTypeFixer.php (hunk 1 above)
Comment (kubawerlos) [reply src/Fixer/FunctionNotation/PhpdocToReturnTypeFixer.php:357]: After rethinking I'm not sure if this is good idea - having a separate class with its own tests seems like an over-engineering - it won't be reusable as `PhpdocToParamTypeFixer` will need a different check (for `void`), so having private method seems fine.

**Pull request merged**